### PR TITLE
Revised CancellationToken to Clarify ExecutionContext/SynchronizationContext Capture Details

### PR DESCRIPTION
**Pull request by bgribaudo**
## Summary

Clarifies when `Register()` captures `ExecutionContext` and `SynchronizationContext`. 

Of particular note, this PR fixes an error where the documentation incorrectly states that `Register()`'s `useSynchronizationContext` parameter controls the capturing of `ExecutionContext` when it actually controls capturing of `SynchronizationContext`.

## Details

Details on how and when these two contexts are captured can be seen in [CancellationToken.cs](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Threading/CancellationToken.cs/). 

In the source, notice how all four public `Register()` methods are hardcoded to `useExecutionContext: true` (indicating that `ExecutionContext` is always captured) while the two that accept parameter `useSynchronizationContext` use that value to control capturing of SynchronizationContext. 

## Suggested Reviewers

@stephentoub

### CODE REVIEW

File: xml/System.Threading/CancellationToken.xml [hunk 1]
```diff
@@ -510,8 +510,10 @@ For the definition of equality, see the <xref:System.Threading.CancellationToken
 ## Remarks  
  If this token is already in the canceled state, the delegate will be run immediately and synchronously. Any exception the delegate generates will be propagated out of this method call.  
   
- The current <xref:System.Threading.ExecutionContext>, if one exists, will be captured along with the delegate and will be used when executing it.  
-  
+ The current <xref:System.Threading.ExecutionContext> is captured along with the delegate and will be used when executing it. 
+
+ The current <xref:System.Threading.SynchronizationContext> is not captured. When the delegate is executed, the <xref:System.Threading.SynchronizationContext> in use at that time, if one exists, is used.
```
Comment (stephentoub) [inline xml/System.Threading/CancellationToken.xml]: > When the delegate is executed, the <xref:System.Threading.SynchronizationContext> in use at that time, if one exists, is used.

This is misleading and should be removed from both here and the other occurrences below.  When useSynchronizationContext is true, we grab SynchronizationContext.Current, and then the invocation of the delegate is done by passing it to the SynchronizationContext for it to invoke.  If useSynchronizationContext is false, or if an overload is used that doesn't take useSynchronizationContext, then SynchronizationContext doesn't play into the equation at all, so nothing "at that time, if one exists, is used"... no SynchronizationContext is used in that case.

**Review by stephentoub — COMMENTED** [review]
Thanks for the fix.  One comment for something to be changed, otherwise LGTM.

**Review by BillWagner — APPROVED** [review]
Thanks for spotting this an fixing it @bgribaudo 
I've reviewed the changes, including the updates per @stephentoub 's comments.  I'll :shipit: once the build finishes.  It should be live on docs in the next day or two on our regular publish cycle.

Thanks again!

**Comment by bgribaudo** [comment]
Thanks, @BillWagner and @stephentoub!

**Comment by bgribaudo** [comment]
@BillWagner, would it be worthwhile for me to backport these changes into CancellationToken.cs's XML documentation comments? Looks like the error this PR fixed is present there, as well.

**Comment by BillWagner** [comment]
@bgribaudo That's a question for @stephentoub  The (very short) version is that we extract the comments from the source and then the API docs and the source are separate. The reason is that there is often more than 1 source stream (.NET Framework, .NET Core, Xamarin, Mono) So we keep the API docs separate.

**Comment by mairaw** [comment]
@karelz to comment about fixing wrong source comments or not.

**Comment by karelz** [comment]
@bgribaudo if you want to fix the XML comments in CoreFX, that is fine. It is likely less used than docs / IntelliSense though. And it will likely never flow back into docs / IntelliSense. (just setting expectations)

**Pull request merged**